src/Displaz.jl

```diff
@@ -286,6 +288,10 @@ function plot3d(plotobj::DisplazWindow, position; color=[1,1,1], markersize=[0.1
     if size(color,2) == 1
         color = repmat(color, 1, nvertices)
     end
+
+    extra_fields = map(x -> (size(x[2]) == (nvertices,) || error("extra fields must be vectors of same length as number of points in position array") ;
+                                        (x[1], array_semantic, map(Float32, x[2]).')), kwargs) # works for vectors only at this stage...
```

Comment:
@andyferris this would actually be a perfect use of the `do` syntax:

```julia
extra_fields = map(kwargs) do arg
    name, value = arg
    size(value) == (nvertices,) || error("extra fields must be vectors of same length as number of points in position array")
    (name, array_semantic, map(Float32, value).')
end
```